### PR TITLE
chore: add scripts to update discovery artifacts

### PR DESCRIPTION
**Pull request by parthea**
These PR add the scripts from #1187 that are needed to update discovery artifacts using a Github action. The scripts will be removed from #1187 once all of the review comments from #1187 have been resolved.

This PR adds the following files under the `scripts/` folder
- `README.md` to provide instructions on manually updating discovery artifacts and API reference documentation.
- `buildprbody.py` creates a summary of the changes detected in discovery artifacts and writes them to `allapis.summary`.
- `changesummary.py` creates verbose change information for each API with discovery artifact changes.
- `createcommits.sh` creates git commits for each API with discovery artifact changes or reference document changes.
- `updatediscoveryartifacts.py` is the python file that can be used to update discovery artifacts.

I also moved `describe.py` under the scripts folder and modified it to save the discovery artifacts that are fetched.

TODO:
- [x] Add tests for scripts
- [x] Address review comments in #1187

### CODE REVIEW

**Comment by parthea** [comment]
@busunkim96 @tswast This is ready for review. PTAL.

**Review by tswast — APPROVED** [review]
Looks good! A few nits re: pathlib vs os, but feel free to ignore / stick with os.

File: scripts/changesummary.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+# limitations under the License.
+
+from enum import IntEnum
+import json
+from multiprocessing import Pool
+import os
+import pandas as pd
+import numpy as np
+
+BRANCH_ARTIFACTS_DIR = "googleapiclient/discovery_cache/documents/"
```
[the remark this answers is not in the document]
Comment (parthea) [reply scripts/changesummary.py]: Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/8c25e1fff7d7209603ab0f19c1bf931834051a47

File: noxfile.py [hunk 2]
```diff
@@ -75,3 +76,21 @@ def unit(session, oauth2client):
         "tests",
         *session.posargs,
     )
+
+@nox.session(python=["3.9"])
+def scripts(session):
+    session.install(*test_dependencies)
+    session.install(".")
```
[the remark this answers is not in the document]
Comment (parthea) [reply noxfile.py]: Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/8c25e1fff7d7209603ab0f19c1bf931834051a47

File: describe.py [hunk 3]
```diff
@@ -429,48 +458,55 @@ def document_api_from_discovery_document(uri):
...
+    for all apis listed in the public discovery directory.
+    args:
+        directory_uri (str): uri of the public discovery directory.
+        doc_destination_dir (str): relative path where the reference
+            documentation should be saved.
+    """
+    api_directory = collections.defaultdict(list)
+    http = build_http()
+    resp, content = http.request(
+        directory_uri, headers={"X-User-IP": "0.0.0.0"}
```
[the remark this answers is not in the document]
Comment (parthea) [reply describe.py]: Removed `X-User-IP` in https://github.com/googleapis/google-api-python-client/pull/1286/commits/fc52778164040644d677080fb2bd7086b5e4db14

File: describe.py [hunk 4]
```diff
@@ -383,44 +380,76 @@ def document_collection_recursive(resource, path, root_discovery, discovery):
...
+        )
+    )
+
+    if resp.status == 200:
+        discovery = json.loads(content)
+        service = build_from_document(discovery)
+        version = safe_version(version)
+        doc_name = "{}.{}.json".format(name, version.replace("_", ""))
+
+        discovery_file_path = os.path.join(DISCOVERY_DOC_DIR, doc_name)
```
[the remark this answers is not in the document]
Comment (parthea) [reply describe.py]: Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/8c25e1fff7d7209603ab0f19c1bf931834051a47

File: describe.py [hunk 5]
```diff
@@ -133,7 +136,8 @@
   <code><a href="#$name">$name($params)</a></code></p>
 <p class="firstline">$firstline</p>"""
 
-BASE = "docs/dyn"
+BASE = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                        "docs/dyn")
```
[the remark this answers is not in the document]
Comment (parthea) [reply describe.py]: Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/8c25e1fff7d7209603ab0f19c1bf931834051a47

File: describe.py [hunk 6]
```diff
@@ -37,12 +38,14 @@
...
 from googleapiclient.discovery import build_from_document
 from googleapiclient.discovery import UnknownApiNameOrVersion
-from googleapiclient.discovery_cache import get_static_doc
 from googleapiclient.http import build_http
 from googleapiclient.errors import HttpError
 
 import uritemplate
 
+DISCOVERY_DOC_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                    'googleapiclient/discovery_cache/documents')
```
[the remark this answers is not in the document]
Comment (parthea) [reply describe.py]: Fixed in https://github.com/googleapis/google-api-python-client/pull/1286/commits/8c25e1fff7d7209603ab0f19c1bf931834051a47

**Comment by parthea** [comment]
@tswast All comments have been addressed. PTAL

**Review by tswast — APPROVED** [review]
Looks good!

**Pull request merged**